### PR TITLE
TST: only test f2py, not f2py2.7 etc, fixes #6718

### LINKED ISSUE #6718: f2py test fail in pip installed numpy-1.10.1 in virtualenv
running `python -c 'import numpy ; numpy.test();'` i get the following error about test_f2py:

```python
$ python -c 'import numpy ; numpy.test();'
Running unit tests for numpy
NumPy version 1.10.1
NumPy relaxed strides checking option: True
NumPy is installed in /usr/local/lib/python2.7/site-packages/numpy
Python version 2.7.10 (default, Jul 13 2015, 12:05:58) [GCC 4.2.1 Compatible Apple LLVM 6.1.0 (clang-602.0.53)]
nose version 1.3.7
......................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................S.........................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................K................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................K...SK.S.......S...................................................................................clang: warning: -Wl,-rpath=/var/folders/95/hl31m68d6ks1d8txd27wxq000000gn/T/tmpGRMpd5/libbar.so: 'linker' input unused
..........................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................S..................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................S......................................................................................................................S.....................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................E
======================================================================
ERROR: test_scripts.test_f2py
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/usr/local/lib/python2.7/site-packages/numpy/testing/decorators.py", line 146, in skipper_func
    return f(*args, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/numpy/tests/test_scripts.py", line 68, in test_f2py
    code, stdout, stderr = run_command([f2py_cmd, '-v'])
  File "/usr/local/lib/python2.7/site-packages/numpy/tests/test_scripts.py", line 48, in run_command
    proc = Popen(cmd, stdout=PIPE, stderr=PIPE)
  File "/usr/local/Cellar/python/2.7.10_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/subprocess.py", line 710, in __init__
    errread, errwrite)
  File "/usr/local/Cellar/python/2.7.10_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/subprocess.py", line 1335, in _execute_child
    raise child_exception
OSError: [Errno 2] No such file or directory

----------------------------------------------------------------------
Ran 5919 tests in 35.478s

FAILED (KNOWNFAIL=3, SKIP=7, errors=1)
```

### PR DESCRIPTION
**Pull request by joernhees**
see #6718 for discussion

### CODE REVIEW

**Comment by charris** [comment]
Apparently `f2py` is also missing sometimes. Hmm..., maybe try in order?

**Comment by joernhees** [comment]
i extended the test so it falls back to the old version of the test and only fails if none works...

that said i'm not entirely sure the old version of the test should've passed for the [failing test](https://travis-ci.org/numpy/numpy/builds/94426666) above, which is on python 3.3 but on the environment contains something like PYTHON=3.4... on py3.3 it should've been installed as f2py3.3, on 3.4 as f2py3.4, so something else seems dodgy in there...

i'm happy to squash if desired, but wanted to give the chance for reviewing the failing test above...

File: numpy/tests/test_scripts.py [hunk 1]
```diff
@@ -63,7 +63,18 @@ def test_f2py():
...
+        success = False
+        for f2py_cmd in f2py_cmds:
+            try:
+                code, stdout, stderr = run_command([f2py_cmd, '-v'])
+                assert_equal(stdout.strip(), asbytes('2'))
+                success = True
+                break
+            except FileNotFoundError:
+                pass
+        assert success, "wasn't able to find f2py or %s on commandline" % f2py_cmds[1]
```
Comment (charris) [inline numpy/tests/test_scripts.py]: Use `assert_(...)`, plain old assert goes away when python runs optimized.

Comment (charris) [inline numpy/tests/test_scripts.py]: You will need to import it from `numpy.testing`.

**Comment by joernhees** [comment]
thanks, done... shall i squash?

**Comment by charris** [comment]
Thanks @joernhees .

**Pull request merged**